platform/src/main/java/org/stellar/anchor/platform/payment/observer/stellar/PaymentObservingAccountsManager.java

```diff
@@ -0,0 +1,157 @@
+    scheduler.scheduleAtFixedRate(
+        this::evictAndPersist, 60, getEvictPeriod().getSeconds(), TimeUnit.SECONDS);
+  }
+
+  public void evictAndPersist() {
+    Log.info("Evicting old accounts...");
+    this.evict(getEvictMaxAge());
+    Log.info("Persisting accounts...");
+    for (ObservingAccount account : this.getAccounts()) {
+      store.upsert(account.account, account.lastObserved);
+    }
+  }
+
+  /**
+   * Adds an account to be observed. If the account is being observed, it will be updated.
+   *
+   * @param account The account being observed.
+   * @param expiring true if the account can be expired. false, otherwise.
+   */
+  public void upsert(String account, Boolean expiring) throws ValueValidationException {
+    if (account != null && expiring != null) {
+      upsert(new ObservingAccount(account, Instant.now(), expiring));
+    }
+  }
+
+  /**
+   * Add an account to be observed. If the account is being observed, it will be updated.
+   *
+   * @param observingAccount The account being observed.
+   */
+  public void upsert(ObservingAccount observingAccount) throws ValueValidationException {
+    if (observingAccount != null) {
+      ObservingAccount existingAccount = allAccounts.get(observingAccount.account);
+      if (existingAccount == null) {
+        allAccounts.put(observingAccount.account, observingAccount);
+        // update the database
+        store.upsert(observingAccount.account, observingAccount.lastObserved);
+      } else {
+        if (!observingAccount.expiring.equals(existingAccount.expiring))
+          throw new ValueValidationException(
+              String.format(
+                  "The expiring flag cannot be modified. Account=[%s]", observingAccount.account));
+        existingAccount.account = observingAccount.account;
+        existingAccount.lastObserved = observingAccount.lastObserved;
+      }
+    }
+  }
+
+  /**
+   * Remove the account from the observed account list.
+   *
+   * @param account The account being removed.
+   */
+  public void remove(String account) {
+    if (account != null) {
+      ObservingAccount targetAccount = allAccounts.get(account);
+      if (targetAccount != null) {
+        allAccounts.remove(account);
+      }
+    }
+  }
+
+  /**
+   * Gets the list of observed accounts.
+   *
+   * @return The list of observed accounts.
+   */
+  public List<ObservingAccount> getAccounts() {
+    return new ArrayList<>(allAccounts.values());
+  }
+
+  /**
+   * Check if the account is being observed. If the account is being observed, the lastObserved
+   * timestamp of the observing account will be updated.
+   *
+   * @param account The account to be checked.
+   * @return true if the account is being observed. false, otherwise.
+   */
+  public boolean observe(String account) {
+    ObservingAccount acct = allAccounts.get(account);
+    if (acct == null) return false;
+    acct.lastObserved = Instant.now();
+    return true;
+  }
```

Comment:
🤔  A method named `observe` would make more sense if we had an **observer** receiving/observing an **event**, but what you're doing here is updating a field rather than observing an event.

What about `isObserving`?

Reply:
I had been struggling with this name. `isObserving` implies this function is read-only. But we are actualy updating the `lastObserved` field. 

This is the payment observer to `observe` if the account is being followed, monitored, or observed.

I agree `observe` is not perfect. But choose between `isObserving` and `observe`, I would prefer `observe`.

Reply:
If this was a private method I'd roll with `observe` but since it's used in other classes (and it's part of the SDK) I still think we should use different wording. Do any of these options appeal to you?
* canObserve
* refreshObservedAccountTimestamp
* tryRefreshingObservedAccountTimestamp
* tryUpdatingLastObservedField
* tryUpdatingLastObservedTimestamp
* elicitObservedAccountTimestamp
* elicitObservedAccount
* pingObservedAccount
* ping

Reply:
Renamed to `lookupAndUpdate`